### ISSUE TITLE
fix rrdesi_mpi aborts

### ISSUE THREAD

**Issue by sbailey** [desihub/redrock#100]
Spinning off a new ticket from #77:

`rrdesi_mpi` when run on dc17b will go for 2 hours and then die with an "Aborted" message:
```
salloc: Nodes nid00[087-094,113,116-127,137-160,168-182] are ready for job
<cori dc17b> date
Tue Jan 23 10:56:50 PST 2018
<cori dc17b> srun --cpu_bind=cores -n 1920 -c 2 desi_pipe_run_mpi --first redshift --last redshift
each proc will wait a few seconds before starting, max is 19.2 sec
INFO:pipe_run.py:60:main: starting at Tue Jan 23 10:57:27 2018
INFO:pipe_run.py:61:main: using raw dir /global/cscratch1/sd/sjbailey/desi/dc17b/spectro/redux/dc17b
INFO:pipe_run.py:62:main: using spectro production dir /global/cscratch1/sd/sjbailey/desi/dc17b/spectro/redux/dc17b
INFO:run.py:499:run_steps: running steps redshift to redshift
INFO:run.py:519:run_steps: starting step redshift at Tue Jan 23 11:04:48 2018
srun: error: nid00160: tasks 1408-1418,1420,1422-1423: Aborted
srun: Terminating job step 9696774.1
srun: error: nid00160: tasks 1419,1421: Aborted
slurmstepd: error: *** STEP 9696774.1 ON nid00087 CANCELLED AT 2018-01-23T13:08:46 ***
srun: error: nid00160: tasks 1424-1432,1436,1439: Terminated
srun: error: nid00181: tasks 1856,1858-1860,1862-1863,1865-1871,1874-1876,1879,1883,1887: Terminated
...
srun: Force Terminated job step 9696774.1
```
Other notes:
1. for the specfiles that did *not* have zbest output, none of them had log files from the pipeline.  Even if one rank was unceremoniously killed, I would have expected partial logfiles to be left over from the other ranks.  I don't know where those logs went.
2. For the logfiles that remained (and had zbest output), none of them had "Terminated", "Aborted", or "memory", or "oom" in them.
3. NERSC staff confirmed that the process received a signal 15, which is different from slurm killing it due to memory problems.  We don't know what sent that signal or why.  They suggested adding a signal handler to catch what rank is receiving the message and what step in the code, and then contacting them again to help debug further.

Unfortunately the problem seems rare enough that we have to run for hours to hit it, but it is common enough to be problematic for using `rrdesi_mpi` as the default method.  In the meantime we can use the wrap-redrock MPI+multiprocessing wrapper, but the MPI version is faster and is supposed to work better at NERSC.

**Comment by sbailey** [comment]
This problem might be caused by the indexing overflow in trapz_rebin reported in #120.  Fix that first, and then retest whether this abort still occurs.

**Comment by moustakas** [comment]
This ticket is nearly six years old and I'm not sure it's an issue anymore; let's reopen (or open a new ticket) if so.
